hip/components/memory.hip.hpp

```diff
@@ -6,6 +6,7 @@
 #define GKO_HIP_COMPONENTS_MEMORY_HIP_HPP_
 
 
+#include <cstring>
```

Comment:
is it used?

Reply:
`memcpy` is defined in string.h, not sure if this is actually necessary though